jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -259,6 +264,16 @@ private Span setTagAsObject(String key, Object value) {
     return this;
   }
 
+  /**
+   * uses {@link #setTagAsObject} to only insert tags that don't have special significance
+   */
+  private void sanitizeAndSetTags(Map<String, Object> tags) {
+    this.tags = new HashMap<>();
+    for (Map.Entry<String, Object> tag : tags.entrySet()) {
+      setTagAsObject(tag.getKey(), tag.getValue());
```

Comment:
a little worried about performance here - setTagAsObject locks every time. if sanitizeAndSetTags is only called in the constructor, I don't think we need to worry about syncing.